core/src/test/java/org/elasticsearch/index/mapper/geo/GeoPointFieldMapperTests.java

```diff
@@ -282,6 +283,42 @@ public void testValidateLatLonValues() throws Exception {
         } catch (MapperParsingException e) {
 
         }
+
+        try {
+            defaultMapper.parse("test", "type", "1", XContentFactory.jsonBuilder()
+                    .startObject()
+                    .startObject("point").field("lat", "-").field("lon", 1.3).endObject()
+                    .endObject()
+                    .bytes());
+            fail();
+        } catch (MapperParsingException e) {
+            assertTrue(e.getRootCause() instanceof NumberFormatException);
```

Comment:
Can you use the dedicated `instanceOf` matcher instead of `assertTrue`? So, `assertThat` with `instanceOf`? The advantage of using the dedicated matchers is that it gives very nice error messages when the assertion fails (so with `assertTrue` all that we see when a test fails is something like "expected true but was false" and with the `assertThat` and a dedicated matcher something like "expected an instance of NumberFormatException but was IllegalArgumentException").